dts/bindings/ethernet/litex,eth0.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2019 Antmicro <www.antmicro.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: LiteX Ethernet
+version: 0.1
+
+description: >
+    This binding gives a base representation of LiteX Ethernet
+
```

Comment:
This should inherit from either `base.yaml` or `ethernet.yaml`, like the other drivers.

If you inherit from `ethernet.yaml` you can provide the mac address in device tree, rather than six Kconfig variables.

Reply:
Inheriting is a good idea. For now I went with `base.yaml` and want to disucss the second option.

I understand that when using `ethernet.yaml` I should put some default MAC address in the device tree and add a choice option in Kconfig to select the source of MAC address.

I see three possible sources:

* random (preferably the default), where:
    * 3 upper bytes are taken from - here comes the first question - Kconfig/device tree
    * 3 lower bytes are randomized,
* manual, where all 6 bytes are taken from the Kconfig,
* the default one, where all 6 bytes are taken from the device tree.

What is the actual difference between options 2 and 3? Assuming I have the default values in Kconfig the same as in the device tree, the 3rd option seems redundant to me.

Reply:
I can't tell you how to do it, because there is no standard.  Two of the existing drivers hard-code the OUI (first three bytes) in the driver, then take the NIC part (last three bytes) from device tree or randomization or elsewhere.  This was probably historical: presumably originally the NIC came from Kconfig then when devicetree support was added it went there.

I personally would assign it in devicetree with an appropriate OUI and default the NIC part to something obviously unassigned like `FF FF FF`, then allow Kconfig to specify that the NIC part could be overwritten with a random value or a value extracted from a device-specific ID (the `hwinfo` API).  It's easy enough to use a devicetree overlay to customize the value, so I wouldn't bother specifying the NIC value through Kconfig.

Somebody else might feel strongly that another way is better.

Reply:
Using devicetree overlay solves the problem of setting the address manualy in an elegant way, so I removed the Kconfig option.
There is no device-specific ID in liteeth, so `hwinfo` driver will not help here.

Summing up:
* by default the NIC part of MAC address will be randomized (although support for random is limitted as mentioned [here](https://github.com/zephyrproject-rtos/zephyr/pull/17462#discussion_r302584580)),
* there is a Kconfig option to turn randomization off and use devicetree overlay to set custom MAC address

Reply:
@pagbigot Do you have any other comments/suggestions?

Reply:
> @pabigot Do you have any other comments/suggestions?

No, the device tree part is fine.  I would have left randomization defaulted off, but existing practice is inconsistent and STM32 seems to support default on, so it's fine.

I don't plan to review the rest of the code; I'm not familiar with this driver system.

Reply:
Ok, thanks for the input! I mark this conversation as resolved.